mobile/src/home-view.js

```diff
@@ -155,6 +155,19 @@ export default class HomeView extends Component {
     .catch(error => this.setState({questionError: "Retry"}))
   }
 
+  trimWhiteSpaceItem = () => {
+    let editingItem = this.state.currentItem
```

Comment:
Even though you are adding properties to `editingItem` you can use `const` instead of `let`, because you are not reassigning `editingItem = ...`.  However, adding properties to `editingItem` is actually (inadvertently?) modifying `this.state.currentItem` (without a `this.setState()` call) since both point to the same object.  Since you probably don't intent do modify state here, I would create a copy of `this.state.currentItem`, e.g.

```javascript
const {currentItem} = this.state
const editingItem = {
  ...currentItem,
  description: currentItem.description.trim(),
  lastLocation: currentItem.lastLocation.trim(),
  currentLocation: currentItem.type === 'found' ? currentItem.currentLocation.trim() : undefined
}
```